### DISCUSSION TITLE
Layout Horizontal Question

### DISCUSSION THREAD

**Discussion by janstadt** [tabler/tabler#1956] [Q&A]
Im evaluating tabler as an addition to my companies ui component framework (based on bootstrap) and noticed that while in the `layout-horizontal` layout there are a few items in the navbar that i cannot seem to find when the viewport gets smaller. Where do these items go and are they able to be displayed always:
<img width="1451" alt="image" src="https://github.com/user-attachments/assets/9a6749e7-c987-41e4-9faa-82c039cc822c">

smaller viewport:
<img width="386" alt="image" src="https://github.com/user-attachments/assets/e75bcef9-3d65-43a7-ac16-a09ca4c7abca">

**Comment by janstadt** [comment]
Anyone have any ideas?

**Comment by janstadt** [comment]
Still would love to know how this all works. I am in the process of pitching tabler to my company for a few UI updates and it would be great to ensure that all our bases are covered.

**Comment by ethancrawford** [comment]
@janstadt you may have moved on from this by now, but if you were still wondering:
> where do these items go

 These items, (or at least their top level elements) have the `d-none` class applied, to hide those parts once a certain smaller size is reached - they're still there, they just disappear. 
 
 >  are they able to be displayed always
 
They can of course be _made_ to do so, with changes to the html/css code. Whether it is of interest to the project maintainers to do this for the navigation on tabler's preview/demo site is another question... however, for your own implementation, at minimum you'd need to remove those `d-none` classes. That would get you some of the way there - the css classes on the theme switcher and notification icons would need slight adjustments too to prevent them from displaying out of alignment.
As for the search input, that is more fiddly because it sits in the same nav bar as the navigation that displays horizontally, so you can't just force the search input to be visible as it would also force the horizontal navigation items to be visible. You'd likely need to extract the search form into the other (or an entirely new) nav bar.

**Comment by janstadt** [comment]
@ethancrawford yes im aware they're still there and i know how to make them appear, i think it is important though to identify why they're being hidden (unsure why they would be hidden for the demo) as folks may want those things to display when in a smaller viewport. Some additional documentation stating that a subset of nav-menu's may be hidden when in a smaller viewport in their docs even would be sufficient. Im pitching this toolkit to my company and want to be certain that i understand its limitations as they more than certainly would have questions about something like this.
